### ISSUE TITLE
2nd SD Card Partition Issues

### ISSUE THREAD

**Issue by Yowlen** [arpitkh96/AmazeFileManager#345]
I use Link2SD, which mounts the 2nd SD Card partition to /data/sdext2. Amaze seems to remount that partition to read-only whenever I try to modify it. Copying/moving/etc. works fine in the terminal and other root file managers, so it's definitely an issue with Amaze. The weirdest part is that Amaze can see the files when I browse the partition, but never ends up modifying anything beyond creating folders, no matter whether copying to or from the partition. It's a standard EXT4 partition, same as what Android usually uses for its internal system, and copying to/from /data or /system works without a hitch.

I know this worked fine at some point in the past, & after some brief testing, it's definitely bugged in 3.0.1+. I didn't test earlier versions.

Edit: It seems the partition remounts to read-only when writing to the partition, but not when reading from it.

**Comment by vishal0071** [comment]
Test on v3.1.2 beta 9

**Comment by Yowlen** [comment]
It stalls completely now. The partition no longer goes to read-only, but the notification stays at 0.00B/X.XX copied/moved, and nothing is actually created or removed. Not even folders get created when doing folder copies.

**Comment by vishal0071** [comment]
Maybe partition is not allowed to remount as rw then? Did you successfully run command on a terminal? If you did, turn what command did you use to remount.

**Comment by Yowlen** [comment]
That's not the problem. The initial problem was that Amaze was remounting it as *read-only* when it shouldn't have been remounting it to begin with because the partition was already read-write. That doesn't happen in the latest beta anymore.

In the latest beta, it stays mounted as read-write (the `mount` command output proves that), but Amaze doesn't actually do the operations I requested. I tested whether it stayed read-write by actually doing a `cp` command in the terminal on the exact folder I tried to copy in Amaze. It worked flawlessly.

And just to reiterate, as I stated in the original post, `/data/sdext2` is the secondary partition (usually EXT4 like the internal Android filesystem) of the microSD card which apps like `Apps2SD` & `Link2SD` on the Play Store create & use to move apps to save space on the internal storage. If it were not allowed to be writable, apps moved there would be unable to write their settings to it, causing crashes & a lot of other havoc.
